### PR TITLE
Update RPM spec file, add xdmod user/group

### PR DESCRIPTION
**Pull request by jtpalmer**
## Description

Changes RPM spec file to create an xdmod user and group.  Updates permissions and ownership of `portal_settings.ini`.  Updates documentation.

## Motivation and Context

To improve security of files containing passwords.

## Tests performed

Built and installed new RPM.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: open_xdmod/modules/xdmod/xdmod.spec.in [hunk 1]
```diff
@@ -29,6 +29,12 @@ detailed interactive charts, graphs, and tables.
 %prep
 %setup -q -n %{name}-%{version}__PRERELEASE__
 
+%pre
+getent group xdmod >/dev/null || groupadd -r xdmod
+getent passwd xdmod >/dev/null || \
+    useradd -r -M -c "Open XDMoD" -g xdmod -d %{_localstatedir}/lib/%{name} -s /sbin/nologin xdmod
+exit 0
```
Comment (jpwhite4) [inline open_xdmod/modules/xdmod/xdmod.spec.in]: Is the `exit 0` needed?

**Comment by jpwhite4** [comment]
We should do this for the SUPReMM module too since the mongo credentials will be in portal_settings.d/supremm.ini (if they have enabled auth for mongo).

File: open_xdmod/modules/xdmod/xdmod.spec.in [hunk 2]
```diff
@@ -61,9 +73,14 @@ rm -rf $RPM_BUILD_ROOT
 %{_datadir}/%{name}/
 %{_docdir}/%{name}-%{version}__PRERELEASE__/
 
-%attr(0775,root,apache) %dir %{_localstatedir}/log/%{name}
+%dir %attr(0775,apache,xdmod) %{_localstatedir}/log/%{name}
 
-%config(noreplace) %{_sysconfdir}/%{name}/
+%dir %{_sysconfdir}/%{name}
+%config(noreplace) %attr(0440,apache,xdmod) %{_sysconfdir}/%{name}/portal_settings.ini
```
Comment (jpwhite4) [inline open_xdmod/modules/xdmod/xdmod.spec.in]: Out of interest what happens here if you do an upgrade - does it change the ownership of the existing file or does it leave it alone?

File: open_xdmod/modules/xdmod/xdmod.spec.in [hunk 3]
```diff
@@ -44,6 +50,12 @@ DESTDIR=$RPM_BUILD_ROOT ./install \
     --crondconfdir=%{_sysconfdir}/cron.d
 
 %post
+# Create log files that need to be writable by both Apache and Open XDMoD scripts.
+for file in exceptions.log query.log; do
+    touch %{_localstatedir}/log/%{name}/$file
+    chown apache:xdmod %{_localstatedir}/log/%{name}/$file
```
Comment (jpwhite4) [inline open_xdmod/modules/xdmod/xdmod.spec.in]: When we do the release notes, I suggest that we should explicitly mention that the log file permission and ownership are modified by the rpm - old log files that were previously readable may now be inaccessible.

File: open_xdmod/modules/xdmod/xdmod.spec.in (hunk 1 above)
Comment (jtpalmer) [reply open_xdmod/modules/xdmod/xdmod.spec.in]: The `exit 0` indicates that the `%pre` section completed successfully even if creation of the group or user failed.  This is recommended here: https://fedoraproject.org/wiki/Packaging:UsersAndGroups?rd=Packaging/UsersAndGroups

(It's also what all the other spec files I've looked at do.)

File: open_xdmod/modules/xdmod/xdmod.spec.in (hunk 3 above)
Comment (jtpalmer) [reply open_xdmod/modules/xdmod/xdmod.spec.in]: I was thinking about not changed the permissions if the files already exist, do you think that would be preferable?

File: open_xdmod/modules/xdmod/xdmod.spec.in (hunk 2 above)
Comment (jtpalmer) [reply open_xdmod/modules/xdmod/xdmod.spec.in]: Assuming you've modified `portal_settings.ini` before upgrading, the ownership and mode does not change.  A file named `portal_settings.ini.rpmnew` is created with the new ownership and mode.

File: open_xdmod/modules/xdmod/xdmod.spec.in (hunk 3 above)
Comment (jpwhite4) [reply open_xdmod/modules/xdmod/xdmod.spec.in]: I'm fine with changing the permissions to make them more restrictive, just wanted to head off any potential support requests.

**Pull request merged**